src/modm/platform/uart/stm32/uart_hal_impl.hpp.in

```diff
@@ -28,45 +35,56 @@ namespace modm::platform
 
 // ----------------------------------------------------------------------------
 void
-{{ name }}::enable()
+{{ hal }}::enable()
 {
-	Rcc::enable<Peripheral::{{ uart_name ~ id }}>();
+	Rcc::enable<Peripheral::{{ name }}>();
 }
 
 void
-{{ name }}::disable()
+{{ hal }}::disable()
 {
 	// TX, RX, Uart, etc. Disable
 	{{ peripheral }}->CR1 = 0;
-	Rcc::disable<Peripheral::{{ uart_name ~ id }}>();
+	Rcc::disable<Peripheral::{{ name }}>();
 }
 
 void
-{{ name }}::enableOperation()
+{{ hal }}::enableOperation()
 {
 	{{ peripheral }}->CR1 |= USART_CR1_UE;
 }
 
 void
-{{ name }}::disableOperation()
+{{ hal }}::disableOperation()
 {
 	{{ peripheral }}->CR1 &= ~USART_CR1_UE;
 }
 
 template< class SystemClock, modm::baudrate_t baudrate, modm::percent_t tolerance >
 void
-{{ name }}::initialize(Parity parity, WordLength length)
+{{ hal }}::initialize(Parity parity, WordLength length)
 {
 	enable();
 	disableOperation();
 
 %% if over8
-	constexpr uint32_t scalar = (baudrate * 16l > SystemClock::{{ uart_name ~ id }}) ? 8 : 16;
+	constexpr uint32_t scalar = (baudrate * 16l > SystemClock::{{ name }}) ? 8 : 16;
```

Comment:
What is the `16l` long literal for? Long and int are both 32bit, in fact `uint32_t` should equal to `unsigned long` for 32 bit arm-none-eabi-gcc. `ull` would be 64 bit

Reply:
Yeah ... not my code 😆 
`16l` does not look useful to me either, so I removed it.